reservation_units/models.py

```diff
@@ -657,6 +659,40 @@ def state(self) -> ReservationUnitState:
 
         return ReservationUnitStateHelper.get_state(self)
 
+    def _requires_product_mapping_update(self):
+        if self.payment_merchant is None:
+            return False
+
+        if self.payment_product is not None or (
+            not self.is_draft
+            and self.pricings.filter(pricing_type=PricingType.PAID)
+            .exclude(status=PricingStatus.PRICING_STATUS_PAST)
+            .exists()
+        ):
+            return True
+
+        return False
+
+    def save(self, *args, **kwargs):
+        # Update product mapping if runit is not draft and is paid
+        if self._requires_product_mapping_update():
+            params = CreateProductParams(
+                namespace=settings.VERKKOKAUPPA_MERCHANT_NAMESPACE,
+                namespace_entity_id=self.pk,
+                merchant_id=self.payment_merchant.id,
+            )
+            api_product = create_product(params)
```

Comment:
this was the part that is to be run in a task?

Reply:
Exactly! I already have the code ready. I just need to update the tests 👍